### PR TITLE
Drop numpy 1.7 and 1.8 from tests and code

### PR DESCRIPTION
**Pull request by mhvk**
This replaces #6003, with all work-arounds for numpy 1.7 and 1.8 removed (only a handful left!).

### CODE REVIEW

**Comment by mhvk** [comment]
I think this gets done in the release notes, but you're right, might as well note it in the changelog, so we don't forget to add it!

File: CHANGES.rst [hunk 1]
```diff
@@ -231,6 +231,7 @@ Other Changes and Additions
 
 - Nothing changed yet.
```
Comment (astrofrog) [inline CHANGES.rst]: This needs to be removed

**Comment by astrofrog** [comment]
Note that ``setup.py`` also needs to be required to enforce the minimum version of Numpy

**Comment by mhvk** [comment]
Thanks!  Also noticed circle is on 1.8, so that needs to change too. I updated things accordingly (I hope there is a 32-bit astropy docker image with 1.9...)

**Comment by mhvk** [comment]
OK, my super-simple change for the docker image failed: how does one get a new one?

**Comment by mhvk** [comment]
OK, all now passes except that for circleci we need a new docker image that has numpy >=1.9 installed. I have no idea how to go about that...

**Comment by bsipocz** [comment]
@astrofrog has the permission to upload a new docker image. What about creating one with a newer numpy that we could also use for v3.0dev? We also need a new one for the affiliated packages as the current one still runs np1.8 with astropy 1.2.1 (https://github.com/astropy/photutils/pull/503#issuecomment-280538077)

**Comment by astrofrog** [comment]
Here is the repository where the docker image can be built:

https://github.com/astropy/astropy-docker/tree/master/32bit

It might be good for someone else to do this to make sure I am not a single point of failure (and I'm happy to add permissions to the docker hub for other people to upload).

**Comment by bsipocz** [comment]
Looking at the system requirements of docker, my laptop is at the edge of being supported (old os, with small RAM and hardly any free space). I could possibly borrow another one when needed, but it seems to make more sense for someone else stepping up for task.

**Comment by mhvk** [comment]
Sorry, don't have time to investigate things beyond what I can easily do in Debian.

**Comment by larrybradley** [comment]
@astrofrog  I can update the docker image, but I need permissions.

**Comment by astrofrog** [comment]
@larrybradley - you should now have permissions

**Comment by larrybradley** [comment]
I uploaded a new docker image with numpy 1.12.1 (the latest release) and the CircleCI test passed.  All is green now.

@astrofrog Can you give me permission to upload the new Dockerfile to `astropy-docker`?  I didn't update the `32bit-with-astropy` image (for affiliated packages).  Should I do that too?

**Comment by mhvk** [comment]
@larrybradley - thanks very much! Since I think there was general agreement on this change, in it goes!

**Comment by astrofrog** [comment]
@larrybradley - you now have permissions to upload the new Dockerfile to astropy-docker, can you do this? I have some other changes I need to make to the Dockerfile, and can regenerate the 32bit-with-astropy one once that's done.

**Comment by larrybradley** [comment]
@astrofrog Done, thanks!

**Pull request merged**